pkg/apply/merge.go

```diff
@@ -219,12 +223,13 @@ func mergeConfigMapForUpdate(current, updated *uns.Unstructured) error {
 		return err
 	}
 
-	for k, v := range s2 {
-		s1[k] += v
-	}
-
-	err = uns.SetNestedStringMap(updated.Object, s1, "data")
-
+	slice1 := strings.Split(s1[AddressPoolConfigMap], "address-pools:")
+	slice2 := strings.Split(s2[AddressPoolConfigMap], "address-pools:")
+	res := append(slice1, slice2...)
+	res = append([]string{"address-pools:"}, res...)
+	data := make(map[string]string)
+	data[AddressPoolConfigMap] = strings.Join(res, "")
+	err = uns.SetNestedStringMap(updated.Object, data, "data")
```

Comment:
cutting and gluing strings feels dangerous, maybe it's better to follow a structured approach: fetch the addresspool string via unstructured.NestedString(upd.Object, "data", "config"), parse them into a slice of locally defined struct , manipulate it and then change the configmap back.

Reply:
+1 on this approach, when changing structures it's better to parse them and render them back

Reply:
I attempt to render the string to data structure and back to string I found myself need to adjust configmap content manually to match with template which I think is pretty ugly , its very possible I am doing it incorrectly here https://play.golang.org/p/u8qpkKATJfZ
if there is anything we can do at the strings themselves better than what I did originally that will be the best , anyway waiting for suggestions

Reply:
I put up a rough draft that seems to work https://gist.github.com/oribon/f23cdcbf4b99c03f9411ecd50c47a0b1

Reply:
Thanks @oribon I didn't use Marshal in the reverse mapping which was wrong , our autoassign magic is acting up now I assume because of *bool type, just one thing there is no manipulation needed just the append is all needed here I will sort out the auto assign and update the diffs, Thanks

Reply:
https://gist.github.com/fedepaol/87f8525ad8fa4dc5a6d422ac575430dd

This should help with the *bool thing I believe. 
Also, I found this https://eli.thegreenplace.net/2020/optional-json-fields-in-go/ . It's about json, but the same applies to yaml.

Reply:
thanks @oribon and @fedepaol for ur help with this one